tests/test-basic-c.c

```diff
@@ -171,6 +172,30 @@ test_raw_file_to_archive_z2_stream (gconstpointer data)
   g_assert_cmpint (checks, >, 0);
 }
 
+static void
+test_repo_exists (gconstpointer data)
+{
+  OstreeRepo *repo = OSTREE_REPO (data);
+  gboolean exists = 0;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GFile) path;
+  int success
+
+  /* Test a return value of TRUE on a repo that exists */
+  success = ostree_repo_exists (repo, &exists, error);
+  g_assert (success && exists);
+
+  /* Test a return value of FALSE on a repo that does not exist */
+  // Create an empty directory
+  mkdir ("not-a-repo", 777);
+  // Create a repo struct associated with that directory
+  path = g_file_new_for_path ("not-a-repo");
+  repo = ostree_repo_new (path);
+  // Check to see if that repo has been initialized in the filesystem (it shouldn't have been)
+  success = ostree_repo_exists (repo, &exists, error);
+  g_assert (success && !exists);
```

Comment:
I guess this is a bit of a nit since it's in a small test app anyway, but for good form: you need to unref the new repo here.